finetune/encoding/target_encoders.py

```diff
@@ -135,6 +136,45 @@ def transform(self, y):
     def inverse_transform(self, y):
         return [self.encoder.decode(y_i.tolist()) for y_i in y]
 
+class SequenceLabelingTextEncoder(Seq2SeqLabelEncoder):
+    def transform(self, y):
+        all_labels = []
+        for y_i in y:
+            # List of dictionaries, where the key is the label and the value is
+            # the span being labeled
+            labels = [{span["label"]: span["text"]} for span in y_i]
+            all_labels.append(json.dumps(labels))
+        ret = super().transform(all_labels)
+        return ret
```

Comment:
This takes a dict as input but returns the strings as predictions? Could we maybe add a TODO here to add the inverse_transform. (or add it?)

Reply:
Good call, had the logic in the target model but putting it as an inverse transform makes much more sense.